oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtGrantedAuthoritiesConverter.java

```diff
@@ -70,8 +72,20 @@ public void setAuthorityPrefix(String authorityPrefix) {
 		this.authorityPrefix = authorityPrefix;
 	}
 
+	/**
+	 * Sets the name of token claim to use for mapping {@link GrantedAuthority authorities} by this converter.
+	 * Defaults to {@link JwtGrantedAuthoritiesConverter#WELL_KNOWN_AUTHORITIES_CLAIM_NAMES}.
+	 *
+	 * @param claimName The token claim name
+	 * @since 5.2
+	 */
+	public void setAuthoritiesClaimName(String claimName) {
+		Assert.hasText(claimName, "claimName cannot be empty");
+		this.claimNames = Collections.singletonList(claimName);
+	}
+
 	private String getAuthoritiesClaimName(Jwt jwt) {
-		for (String claimName : WELL_KNOWN_AUTHORITIES_CLAIM_NAMES) {
+		for (String claimName : this.claimNames) {
```

Comment:
If the application sets the authorities claim name, then there is no need to search for it. Did you run into difficulties with something more like:

```java
if (this.authoritiesClaimName != null) {
    return this.authoritiesClaimName;
}

for (String claimName : WELL_KNOWN...
```

Reply:
No, I  just missed that. There is really no need to search for custom authorities claim. Changed it as suggested.